opencog/reasoning/RuleEngine/rules/pln/evaluation-to-member.scm

```diff
@@ -0,0 +1,64 @@
+; ============================================================================= 
+; GeneralEvaluationToMemberRule
+;	EvaluationLink (pred D (ListLink B C))
+;		becomes
+;	MemberLink( B (SatisfyingSetLink( Variable $X 
+;		(EvaluationLink (pred D (ListLink X C))))))
+; -----------------------------------------------------------------------------
+
+(define pln-rule-evaluation-to-member
+	(BindLink
+		(ListLink
+			(VariableNode "$B")
+			(VariableNode "$C")
+			(VariableNode "$D")
+			(TypedVariableLink
+				(VariableNode "$D")
+				(VariableTypeNode "PredicateNode")))
+		(ImplicationLink
+			(EvaluationLink
+				(VariableNode "$D")
```

Comment:
I started implementing the rule as discussed in the thread but I am running into an issue. [The code](https://gist.github.com/prateeksaxena2809/bae56405a4aed3898088) on compiling gives the following error.
```
ERROR: In procedure cog-outgoing-set: Wrong type argument in position 1 (expecting opencog atom): ()
ABORT: wrong-type-arg
```
I think this is because on compiling, cog-outgoing-set is not able to detect that the (VariableNode "$A") is of type ListLink. If I change [this line](https://gist.github.com/prateeksaxena2809/bae56405a4aed3898088#file-evaluation-to-member-scm-L24)  to (Variable Node "$A") and [these lines](https://gist.github.com/prateeksaxena2809/bae56405a4aed3898088#file-evaluation-to-member-scm-L30-L33)  to (VariableNode "$X") (VariableNode "$A"), the code compiles successfully (but gives conceptually wrong results as expected).

Any ideas on how I could fix the issue?